### ISSUE TITLE
Does this work with fake requests still?

### ISSUE THREAD

**Issue by tannewt** [adafruit/Adafruit_CircuitPython_PortalBase#58]
From a user [on the forum](https://forums.adafruit.com/viewtopic.php?f=60&t=186850):

```
*** USING LOCALFILE FOR DATA - NOT INTERNET!!! ***
Traceback (meest recente call laatst):
  Bestand "code.py", regel 36, in <module>
  Bestand "adafruit_pyportal/__init__.py", regel 310, in fetch
  Bestand "adafruit_portalbase/network.py", regel 509, in check_response
  Bestand "adafruit_portalbase/network.py", regel 535, in _get_headers
AttributeError: 'Fake_Requests' object heeft geen attribuut 'headers'
```

**Comment by dgriswo** [comment]
I don't think it ever did.  @makermelissa initial commit stores the output of Fake_Requests, which is string.  The rest of the library expects response to be an object.

Should Fake_Requests return an object or should PortalBase construct one when Fake_Requests is used?

FWIW, I made it here because the PyPortal library calls PortalBase

**Comment by makermelissa** [comment]
I think it should return an object. Fake Requests was stripped out of the old PyPortal library and I don't think is actually used much, so likely things have changed without Fake Requests being updated properly.

**Comment by Neradoc** [comment]
The `Fake_Requests` class is a response currently. We can make it work by adding a `headers` property and a `status_code` of 200. While helping somebody on discord, I also made it try to load the file as json and set headers["content-type"] to "application/json" or "text/plain" depending on the result. I can PR that to the library. That would cover the basics of testing the user code with a certain kind of content without having to hit the network.

Though I think it would be interesting for debugging purposes to be able to test headers too, maybe with local_headers.txt being a key/value table of headers (and maybe status code or other).

**Comment by dbradmit** [comment]
I just got a [PyPortal](https://www.adafruit.com/product/4116) and it seems like this is still an issue. I have been trying to work through [adafruit-pyportal/parsing-json](https://learn.adafruit.com/adafruit-pyportal/parsing-json) and the code fails when trying to use local.txt due to the headers issue regarding fake requests.

I've spent all day trying to debug or fix it on my own without luck. I am clearly missing something basic though as I can't get a working update to [network.py](https://github.com/adafruit/Adafruit_CircuitPython_PortalBase/blob/main/adafruit_portalbase/network.py). I've tried passing hardcoded headers as well as simply adding print statements to make sure my hardcoded headers match what would be returned by [quotes.php](https://www.adafruit.com/api/quotes.php) but when I compile network.py with mpy-cross and replace the existing network.mpy on my PyPortal with the self-compiled version the updated version never executes.

**Comment by FoamyGuy** [comment]
> but when I compile network.py with mpy-cross and replace the existing network.mpy on my PyPortal with the self-compiled version the updated version never executes.

@dbradmit are you putting the modified network.py into `/lib/adafruit_portalbase/`?

If so I think the reason your modified version isn't executing is that adafruit_portalbase is frozen in to the PyPortal build: https://github.com/adafruit/circuitpython/blob/main/ports/atmel-samd/boards/pyportal/mpconfigboard.mk#L15 

So the frozen in version takes precedent over the one inside of `/lib/`.

If you put your modified one at the root of CIRCUITPY it should take precedence over the frozen one. i.e. your device files would be like:

```
adafruit_portalbase/
lib/
boot_out.txt
code.py
```

**Comment by FoamyGuy** [comment]
resolved by https://github.com/adafruit/Adafruit_CircuitPython_FakeRequests/pull/10
